### PR TITLE
Terminology: "RO-Crate Metadata Document" instead of "Metadata File"

### PR DESCRIPTION
**Pull request by ptsefton**
This actions #247 

This PR touches a lot of files, and is intended to bring the whole spec into line with the new introduction and the the definition of Attached and Detached crates and the recognition that crate metadata can be served by APIs.

The new intro introduced the idea of an "RO-Crate Metadata Document" which is a more abstract version of the term "RO-Crate Metadata File" -- as RO-Crate JSON ld can be moved around without files and the RO-Crate Metadata File is only relevant to to Attached Crates. 

The new intro also renamed the RO-Crate Metadata File descriptor, dropping the "file" but leaving the convention that it has an @id of ro-crate metadata.json.

### CODE REVIEW

File: docs/1.2-DRAFT/metadata.md [hunk 1]
```diff
@@ -60,11 +60,11 @@ RO-Crate makes use of the [Linked Data principles] for its description. In parti
 
 RO-Crate realizes these principles using a particular set of technologies and best practices:
 
-1. The _RO-Crate Metadata File_ and _RO-Crate Website_ can be directly published on the web together with the RO-Crate payload. In addition, a data package (e.g. BagIt Zip archive) that contain the RO-Crate can also be published on the web.
-2. The _RO-Crate Metadata File_ is based on the structured data format [JSON-LD]. 
+1. The _RO-Crate Metadata Document_ can be stored in an _RO-Crate Metadata File_ and an _RO-Crate Website_ can be directly published on the web together with the RO-Crate payload. In addition, a data package (e.g. BagIt Zip archive) that contain the RO-Crate can also be published on the web.
```
Comment (simleo) [inline docs/1.2-DRAFT/metadata.md:63]: The new version does not say that the metadata file can be published on the web anymore. To preserve the original meaning (assuming that was the intention), it should say:

The _RO-Crate Metadata Document_ can be stored in an _RO-Crate Metadata File_; the _RO-Crate Metadata File_ and an _RO-Crate Website_ can be directly published on the web together with the RO-Crate payload.

File: docs/1.2-DRAFT/root-data-entity.md [hunk 2]
```diff
@@ -34,16 +34,13 @@ The **Root Data Entity** is a [Dataset] that represent the RO-Crate as a whole;
 a _Research Object_ that includes the _Data Entities_ and the related
 _Contextual Entities_.
 
-As explained in section [RO-Crate structure](structure.md), the RO-Crate description 
-is stored as _JSON-LD_ in the _RO-Crate Metadata File_ `ro-crate-metadata.json` in 
-the _RO-Crate root_ directory.  In this section we define how the RO-Crate Metadata file
-describes itself and the RO-Crate root.
+An RO-Crate is described using _JSON-LD_ by an _RO-Crate Metadata Document_. As explained in section [RO-Crate structure](structure.md) this may be stored in and _RO-Craate Metadata File_. In this section we describe the format of the JSON-LD document.
```
Comment (simleo) [inline docs/1.2-DRAFT/root-data-entity.md:37]: `in and _RO-Craate Metadata File_` -> `in an _RO-Crate Metadata File_`

File: docs/1.2-DRAFT/structure.md [hunk 3]
```diff
@@ -37,19 +37,19 @@ parent: RO-Crate 1.2-DRAFT
 There are two classes of RO-Crate detailed below:
 
 **Attached RO-Crate**
-: A crate that has a well-defined _RO-Crate Root directory_ and can carry an **explicit payload** of local [data entities](data-entities.md) as regular files (combined with [Web-based Data Entities](data-entities.html#web-based-data-entities) where needed). This type of RO-Crate can be suitable for long-term preservation, transfer and publishing, as the _RO-Crate Metadata File_ is stored alongside the crate's payload. See further definition of [attached RO-Crate](#attached-ro-crate) below.
+: A crate that is present in a file-system context which has an_RO-Crate Root directory_ and can carry an **explicit payload** of local [data entities](data-entities.md) as regular files (combined with [Web-based Data Entities](data-entities.html#web-based-data-entities) where needed). This type of RO-Crate can be suitable for packaging a data payload for long-term preservation, transfer and publishing, as the _RO-Crate Metadata Document_ is stored in an _RO-Crate Metadata File_ alongside the crate's payload. See further definition of [attached RO-Crate](#attached-ro-crate) below.
```
Comment (simleo) [inline docs/1.2-DRAFT/structure.md:40]: `an_RO` -> `an _RO`

File: docs/1.2-DRAFT/structure.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ The payload directory (and its child directory) contains files and directories t
 
 ## Detached RO-Crate
 
-A _Detached RO-Crate_ is an RO-Crate without a defined root directory, where the _RO-Crate Metadata File_ and/or _RO-Crate Website_ content is accessed independently (e.g. as part of a programmatic API).
+A _Detached RO-Crate_ is an RO-Crate without a defined root directory, where the _RO-Crate Metadata Document and/or _RO-Crate Website_ content is accessed independently (e.g. as part of a programmatic API).
```
Comment (simleo) [inline docs/1.2-DRAFT/structure.md:77]: `Document` -> `Document_`

**Review by stain — APPROVED** [review]

**Pull request merged**